File: MapReduce/Surbhi_Batra_task2.py
import pyspark as spark

from pyspark.sql import SparkSession

from pyspark.sql import functions as F

from pyspark.sql.functions import col, avg
from pyspark.sql.types import DoubleType

import sys

infile_ratings = sys.argv[1]
infile_tags = sys.argv[2]
outfile = sys.argv[3]

spark = SparkSession \
		.builder \
		.appName("Task2_try1") \
		.config("spark.some.connfig.option","some-value") \
		.getOrCreate()

# df = spark.read.csv("/home/surbhi/Documents/2018Summer/DM/Assignment_01/ml-latest-small/ratings.csv", header = True)
dfr = spark.read.csv(infile_ratings, header = True, inferSchema= True)
dft = spark.read.csv(infile_tags, header = True, inferSchema= True)
df = dft.join(dfr, dfr.movieId == dft.movieId, 'inner')
df = df.groupBy("tag").agg(F.avg("rating").alias('rating_avg'))
##df = df.select(dft.tag, dfr.rating).collect()
df = df.orderBy(df.tag.desc())
df.repartition(1).write.format("com.databricks.spark.csv").option("header", "true").save(outfile)

# $SPARK_HOME/bin/spark-submit solution/surbhi_batra_task2.py solution/ml-latest-small/ratings.csv solution/ml-latest-small/tags.csv outputFiles/surbhi_batra_result_task2_small.csv
# $SPARK_HOME/bin/spark-submit solution/surbhi_batra_task2.py solution/ml-20m/ratings.csv solution/ml-20m/tags.csv outputFiles/surbhi_batra_result_task2_big.csv

File: MapReduce/Surbhi_Batra_task2.scala
import org.apache.spark
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SparkSession
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.functions.collect_list
import org.apache.spark.sql.functions._

object Task2 {
  def main(args: Array[String]): Unit = {
    val in_ratings = args(0)
    val in_tags = args(1)
    val outfile = args(2)
    //    val conf = new SparkConf().setAppName("Task2").setMaster("local[2]")
    //    val sc = new SparkContext(conf)

    val spark = SparkSession.builder
      .master("local")
      .appName("Spark CSV Reader")
      .getOrCreate;

    val df_ratings = spark.read
      .format("csv")
      .option("header", "true") //reading the headers
      .option("mode", "DROPMALFORMED")
      .option("inferSchema", "True")
      .load(in_ratings)
    //      .load("ml-latest-small/ratings.csv")

    val df_tags = spark.read
      .format("csv")
      .option("header", "true") //reading the headers
      .option("mode", "DROPMALFORMED")
      .option("inferSchema", "True")
      //      .load("ml-latest-small/tags.csv")
      .load(in_tags)

    val res = df_ratings.join(df_tags, usingColumn = "movieId")

    res.groupBy("tag").agg(mean("rating").alias("rating_avg"))
      .orderBy(desc("tag"))
      .coalesce(1)
      .write.format("com.databricks.spark.csv")
      .option("header", "true")
      .save(outfile)
    //      .save("scala_task2_small.csv")

  }
}